### DISCUSSION TITLE
!= operator not working as expected when using other step output

### DISCUSSION THREAD

**Discussion by ureciocais** [community/community#55838] [Actions] [answered]
### Select Topic Area

Bug

### Body

The following examples expect the last echo step to get skipped

This works fine using ==
![Screenshot 2023-05-18 at 2 27 12 PM](https://github.com/community/community/assets/104901556/90b9889f-e6cd-4608-b9ce-aca6f5e6891c)

This doesn't using !=
![Screenshot 2023-05-18 at 2 27 32 PM](https://github.com/community/community/assets/104901556/2d1f2aaa-e911-48a1-b6f4-b6f087eb0e9a)

It works using inputs for instance
![Screenshot 2023-05-18 at 2 52 09 PM](https://github.com/community/community/assets/104901556/39b69dd9-4e7d-45e3-b6bf-867bc1256033)

**Comment by airtower-luna** [comment, marked as answer]
The example with `!=` is correct, the output is unequal to `.`. When you write `varr='.'` to `$GITHUB_OUTPUT` the value of the `varr` output will be `'.'` (including the quotes).

**Reply by ureciocais** [reply]
This is just one example. I tried without quotes and also with `if: '${{  }}' != '.'`
Have you tested on a repo that this works?

**Reply by airtower-luna** [reply]
`if: '${{  }}' != '.'` will *always* be true, because only the part between `${{ ... }}` is evaluated as an expression, the rest is simply a string. And a non-empty string evaluates to true.

And yes, I tested, and the operator [works perfectly fine](https://github.com/airtower-luna/hello-github-actions/actions/runs/5024896766/workflow#L49-L54).

**Reply by ureciocais** [reply]
I see. I understand now. Thanks for explaining that
